### PR TITLE
remove FedoraResource::getNode from fcrepo-kernel-api

### PR DESCRIPTION
**Pull request by acoburn**
Resolves: https://jira.duraspace.org/browse/FCREPO-1869

This will clearly have significant implications on certain downstream projects. If this approach seems sensible, then I will create related PRs for those projects.

### CODE REVIEW

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/FedoraTypesUtils.java [hunk 1]
```diff
@@ -301,4 +302,17 @@ public static Node getClosestExistingAncestor(final Session session, final Strin
...
+    /**
+     * Retrieve the underlying JCR Node from the FedoraResource
+     *
+     * @param resource the Fedora resource
+     * @return the JCR Node
+     */
+    public static Node getJcrNode(final FedoraResource resource) {
+        if (resource instanceof FedoraResourceImpl) {
+            return ((FedoraResourceImpl)resource).getNode();
+        } else {
```
Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/FedoraTypesUtils.java]: Is this `else` needed?

Comment (acoburn) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/FedoraTypesUtils.java]: No, the `else` isn't needed. It's now removed.

**Comment by ajs6f** [comment]
Do you want to create the abovementioned PRs for downstream projects before merging this?

**Comment by acoburn** [comment]
Yes, I would like to create those other PRs before this gets merged. I was mostly waiting for feedback on whether this seems like a good approach.

**Comment by ajs6f** [comment]
Well, I buy it. :+1:

**Comment by acoburn** [comment]
OK, then. More work for me :-)

**Comment by acoburn** [comment]
@ajs6f I submitted PRs for all the downstream projects affected by this.

**Comment by ajs6f** [comment]
I'm going to let this whole thing sit until tomorrow to give everyone a chance to comment, then I'll merge en mass.

**Pull request merged**